File: src/API/src/Modules/Products/Products.Application/ViewModels/ProductLookupBundleViewModel.cs
namespace Products.Application.ViewModels
{
    public class ProductLookupBundleViewModel
    {
        public List<ProductOptionViewModel> ProductTypes { get; set; } = new();
        public List<ProductOptionViewModel> DeploymentTypes { get; set; } = new();
        public List<ProductOptionViewModel> OwnershipTypes { get; set; } = new();
        public List<ProductOwnerPartnerLookupViewModel> OwnerPartners { get; set; } = new();
    }
}

File: src/ClientApp/src/app/pages/products/view-models/product-lookup-bundle.view-model.ts
import { ProductOptionViewModel } from './product-option.view-model';

export interface ProductOwnerPartnerLookupViewModel {
    id: string;
    name: string;
    code?: string | null;
    partnerTypeCode?: string | null;
}

export interface ProductLookupBundleViewModel {
    productTypes: ProductOptionViewModel[];
    deploymentTypes: ProductOptionViewModel[];
    ownershipTypes: ProductOptionViewModel[];
    ownerPartners: ProductOwnerPartnerLookupViewModel[];
}
